packages/calcite-components/src/components/input-time-zone/utils.ts

```diff
@@ -1,43 +1,132 @@
-import { BasicTimeZoneGroup } from "./interfaces";
+import { TimeZoneItem, TimeZoneMode, TimeZoneName } from "./interfaces";
+import { getDateTimeFormat, SupportedLocale } from "../../utils/locale";
+import { InputTimeZoneMessages } from "./assets/input-time-zone/t9n";
 
 const hourToMinutes = 60;
 
+function timeZoneOffsetToDecimal(shortOffsetTimeZoneName: string): string {
+  const minusSign = "−";
+  const hyphen = "-";
+
+  return (
+    shortOffsetTimeZoneName
+      .replace(":15", ".25")
+      .replace(":30", ".5")
+      .replace(":45", ".75")
+
+      // ensures decimal string representation is parseable
+      .replace(minusSign, hyphen)
+  );
+}
+
+function toOffsetValue(timeZoneName: TimeZoneName, referenceDateInMs: number): number {
+  // we use "en-US" to allow us to reliably remove the standard time token
+  const offset = getTimeZoneShortOffset(timeZoneName, "en-US", referenceDateInMs).replace("GMT", "");
+
+  if (offset === "") {
+    return 0;
+  }
+
+  return Number(timeZoneOffsetToDecimal(offset)) * hourToMinutes;
+}
+
 export function getUserTimeZoneOffset(): number {
   const localDate = new Date();
   return localDate.getTimezoneOffset() * -1;
 }
 
-function getFallbackTimeZoneGroups(): BasicTimeZoneGroup[] {
-  const timeZoneOffsets = [
-    -11, -10, -9.5, -9, -8, -7, -6, -5, -4, -3, -2.5, -2, -1, 0, 1, 2, 3, 3.5, 4, 4.5, 5, 6, 6.5, 7, 8, 8.75, 9, 9.5,
-    10, 10.5, 11, 12, 12.75, 13, 14,
-  ];
+export function getUserTimeZoneName(): string {
+  const dateFormatter = new Intl.DateTimeFormat();
+  return dateFormatter.resolvedOptions().timeZone;
+}
+
+/**
+ * The lazy-loaded timezone-groups lib to be used across instances.
+ */
+let timeZoneGroups: Promise<[any, any]>;
 
-  return timeZoneOffsets.map((offset) => {
-    return {
-      offsetValue: offset * hourToMinutes,
-      offsetLabel: toGMTLabel(offset),
-    };
-  });
+export async function createTimeZoneItems(
+  locale: SupportedLocale,
+  messages: InputTimeZoneMessages,
+  mode: TimeZoneMode,
+  referenceDate: Date
+): Promise<TimeZoneItem[]> {
+  const referenceDateInMs: number = referenceDate.getTime();
+  const timeZoneNames = Intl.supportedValuesOf("timeZone");
+
+  if (mode === "offset") {
+    if (!timeZoneGroups) {
+      timeZoneGroups = Promise.all([
+        import("timezone-groups/dist/index.js"),
+        import("timezone-groups/dist/strategy/native/index.js"),
```

Comment:
Is the `timezone-groups` directory being generated in `dist` or somewhere in `src`? I'm kinda curious what these dynamic imports are being compiled to, can you search for one of the paths in `dist` to see if these strings remain intact? If they are, can we be certain the paths will resolve correctly in all of our consumer's different use cases?

Reply:
`timezone-groups` is a new dependency. Lean chunks for each import are created. 🥩

Reply:
Nevermind, I thought we were creating a directory named `timezone-groups` but it's actually an NPM package so we're good.